src/radical/pilot/agent/lm/aprun.py

```diff
@@ -54,9 +54,22 @@ def construct_command(self, cu, launch_script_hop):
         #       'cpu_threads'      : 2,
         #       'gpu_processes     : 2,
         #       'slots':
-        #       {                 # [[nodename, [node_uid], [core indexes],   [gpu idx]]]
-        #         'nodes'         : [[node_1,   node_uid_1, [[0, 2], [4, 6]], [[0]    ]],
-        #                            [node_2,   node_uid_2, [[1, 3], [5, 7]], [[0]    ]]],
+        #       {   # 'nodes': [{'name': node_name,
+        #           #            'uid': node_uid,
+        #           #            'core_map': [core_map],
+        #           #            'gpu_map': [gpu_map],
+        #           #            'lfs': lfs}],
+        #         'nodes'         : [{  'name': node_1,
+        #                               'uid': node_uid_1,
+        #                               'core_map': [[0, 2], [4, 6]],
+        #                               'gpu_map': [[0]],
+        #                               'lfs': 1024},
+        #                            {  'name': node_2,
+        #                               'uid': node_uid_2,
+        #                               'core_map': [[1, 3], [5, 7]],
+        #                               'gpu_map': [[0]],
+        #                               'lfs': 1024}
```

Comment:
replying here on the `task_offsets`: those were not changed by this PR, and only the comments were added, so you should not need to understand what those task offsets are.  For completeness though: instead of an absolute core position, some launch methods require us to specify the position as a relative offset from the start of the allocation, or the start of a node etc - we call those relative location `task_offsets`.

Reply:
Thanks Andre. 
Somebody wrote that due to the changes in this PR,  this code needs to be revisited.
That is why I was not sure about it.

Reply:
Ah, I see, makes sense. But that revision does not need to happen in this PR, fortunately :-)